src/main/java/org/folio/rest/support/EffectiveCallNumberComponentsUtil.java

```diff
@@ -0,0 +1,42 @@
+package org.folio.rest.support;
+
+import org.apache.commons.lang.StringUtils;
+import org.folio.rest.jaxrs.model.EffectiveCallNumberComponents;
+import org.folio.rest.jaxrs.model.HoldingsRecord;
+import org.folio.rest.jaxrs.model.Item;
+
+/**
+ * Utility methods for Effective Call Number Components object.
+ */
+public final class EffectiveCallNumberComponentsUtil {
+
+  private EffectiveCallNumberComponentsUtil() {
+    throw new UnsupportedOperationException("Do not instantiate");
+  }
+
+  public static EffectiveCallNumberComponents buildComponents(HoldingsRecord holdings, Item item) {
+    return updateComponents(new EffectiveCallNumberComponents(), holdings, item);
+  }
+
+  private static EffectiveCallNumberComponents updateComponents(
+    EffectiveCallNumberComponents components, HoldingsRecord holdings, Item item) {
+
+    String updatedCallNumber = null;
+    if (StringUtils.isNotBlank(item.getItemLevelCallNumber())) {
+      updatedCallNumber = item.getItemLevelCallNumber();
+    } else if (StringUtils.isNotBlank(holdings.getCallNumber())) {
+      updatedCallNumber = holdings.getCallNumber();
+    }
+
+    String updatedCallNumberSuffix = null;
+    if (StringUtils.isNotBlank(item.getItemLevelCallNumberSuffix())) {
```

Comment:
Hi @marcjohnson-kint, @mreno-EBSCO, @rladdusaw,

I was reviewing circulation logic and spotted that there we have a bit different logic to for effective suffix/prefix. We have just a single check on `holdings.callNumber` and `item.itemLevelCallNumber` and depending on the result, we pick all the components from either item or holding (not per-component logic). This looks to me more correct, since item and holdings call number may be of different type and mixing them looks incorrect.

Maybe I'm missing something.

Reply:
@bohdan-suprun 

I think that is a question for Cate, please write a comment on the JIRA issue.

My understanding is that we do want to mix the properties at different levels. It is worth confirming that with Cate.

The logic defined in inventory should supersede the interim logic put into circulation by Vega for notices.

Reply:
Hi @rladdusaw, @marcjohnson-kint, Cate responded that we should use per-component logic, so no work observed here.